### PR TITLE
Sign android artifacts

### PR DESCRIPTION
**Pull request by bulbigood**
- Generate signed APK and AAB artifacts for Android with Action.
- Added proguard plugin to minify Android artifacts.
- Fix iOS artifact uploading broken by name changes

Compare artifacts size 
before Proguard: https://github.com/Montelibero/mtl_solar/actions/runs/7877847847
after Proguard: https://github.com/Montelibero/mtl_solar/actions/runs/7882564156

### CODE REVIEW

File: .github/workflows/mobile.yml [hunk 1]
```diff
@@ -19,18 +19,24 @@ jobs:
...
             npm run build:ios
 
         - name: Upload artifacts
-          uses: actions/upload-artifact@v2
+          uses: actions/upload-artifact@v4
           with:
-            name: Solar.app
-            path: /Users/runner/work/mtl_solar/mtl_solar/cordova/platforms/ios/build/emulator/Solar.app
+            name: MTL_Solar_ios.app
+            path: ${{ github.workspace }}/cordova/platforms/ios/build/emulator/Montelibero Solar.app
```
Comment (sondreb) [inline .github/workflows/mobile.yml:38]: Should we avoid having namespace in the filename?

File: .github/workflows/mobile.yml [hunk 2]
```diff
@@ -62,15 +68,39 @@ jobs:
...
+            echo "$ENCRYPTED_KEYSTORE" > "$ENCRYPTED_KEYSTORE_LOCATION"
+            gpg -d --passphrase "$KEYSTORE_GPG_PASSWORD" --batch "$ENCRYPTED_KEYSTORE_LOCATION" > "$KEYSTORE_LOCATION"
+            cd cordova
+            PACKAGE_TYPE=bundle npm run build:android:signed:auto
+            PACKAGE_TYPE=apk npm run build:android:signed:auto
+
+        - name: Upload .apk artifact
+          uses: actions/upload-artifact@v4
+          with:
+            name: MTL_Solar_android.apk
```
Comment (sondreb) [inline .github/workflows/mobile.yml:97]: Should we have the version name in the .apk filename itself?

**Review by sondreb — APPROVED** [review]
Good work, a couple of minor comments.

File: .github/workflows/mobile.yml (hunk 1 above)
Comment (bulbigood) [reply .github/workflows/mobile.yml:38]: What do you mean by "namespace"?

File: .github/workflows/mobile.yml (hunk 2 above)
Comment (bulbigood) [reply .github/workflows/mobile.yml:97]: It could be useful improvement

**Pull request merged**